tests/unittests/cmd/devel/test_net_convert.py

```diff
@@ -0,0 +1,187 @@
+
+[connection]
+id=cloud-init eth0
+uuid=1dd9a779-d327-56e1-8454-c65e2556c12c
+type=ethernet
+interface-name=eth0
+
+[user]
+org.freedesktop.NetworkManager.origin=cloud-init
+
+[ethernet]
+
+[ipv4]
+method=auto
+may-fail=false
+
+"""
+
+
+class TestNetConvert:
+
+    missing_required_args = itertools.combinations(
+        required_args, len(required_args) - 1
+    )
+
+    def _replace_path_args(self, cmd, tmpdir):
+        """Inject tmpdir replacements for parameterize args."""
+        updated_cmd = []
+        for arg in cmd:
+            if arg == "--network-data":
+                net_file = tmpdir.join("net")
+                net_file.write("")
+                updated_cmd.append(f"--network-data={net_file}")
+            elif arg == "--directory":
+                updated_cmd.append(f"--directory={tmpdir.strpath}")
+            else:
+                updated_cmd.append(arg)
+        return updated_cmd
+
+    @pytest.mark.parametrize("cmdargs", missing_required_args)
+    def test_argparse_error_on_missing_args(self, cmdargs, capsys, tmpdir):
+        """Log the appropriate error when required args are missing."""
+        params = self._replace_path_args(cmdargs, tmpdir)
+        with mock.patch("sys.argv", ["net-convert"] + params):
+            with pytest.raises(SystemExit):
+                net_convert.get_parser().parse_args()
+        _out, err = capsys.readouterr()
+        assert "the following arguments are required" in err
+
+    @pytest.mark.parametrize("debug", (False, True))
+    @pytest.mark.parametrize(
+        "output_kind,outfile_content",
+        (
+            (
+                "netplan",
+                {"etc/netplan/50-cloud-init.yaml": SAMPLE_NETPLAN_CONTENT},
+            ),
+            (
+                "eni",
+                {
+                    "etc/network/interfaces.d/50-cloud-init.cfg": SAMPLE_ENI_CONTENT  # noqa: E501
+                },
+            ),
+            (
+                "networkd",
+                {
+                    "etc/systemd/network/10-cloud-init-eth0.network": SAMPLE_NETWORKD_CONTENT  # noqa: E501
+                },
+            ),
+            (
+                "sysconfig",
+                {
+                    "etc/sysconfig/network-scripts/ifcfg-eth0": SAMPLE_SYSCONFIG_CONTENT  # noqa: E501
+                },
+            ),
+            (
+                "network-manager",
+                {
+                    "etc/NetworkManager/system-connections/cloud-init-eth0.nmconnection": SAMPLE_NETWORK_MANAGER_CONTENT  # noqa: E501
+                },
+            ),
+        ),
+    )
+    def test_convert_output_kind_artifacts(
+        self, output_kind, outfile_content, debug, capsys, tmpdir
+    ):
+        """Assert proper output-kind artifacts are written."""
+        network_data = tmpdir.join("network_data")
+        network_data.write(SAMPLE_NET_V1)
+        distro = "fedora" if output_kind == "sysconfig" else "ubuntu"
```

Comment:
Nit: Since fedora just switched to network-manager, maybe we should use a slower-moving distro for the sysconfig distro?